programs/openbook-v2/src/state/open_orders_account.rs

```diff
@@ -187,6 +187,13 @@ impl OpenOrdersAccount {
             }
         }
 
+        // calculate taker fee for logging, ignoring self trades
+        let taker_fees = if quote_native > 0 && fill.maker != fill.taker {
```

Comment:
Please note that this amount will not correctly reflect the fees paid by the taker as it can be (slightly) greater than the actual fees paid: taker fees are computed from the total taken amount, not from the sum of the fill amounts, that is `market.taker_fee_ceil(total_base_taken_wo_self_trade) <= \sum_i market.taker_fee_ceil(fill_amount_wo_self_trade_i)`

Reply:
I don't fully understand this. Wouldn't the difference between incrementally calculating the taker fee vs. calculating using the total taken amount just be attributable to the rounding in `market.taker_fees_ceil()`?

I'd expect that `total_quote_lots_taken` and then `total_quote_taken_native_wo_self` would be equal to the sum over all fill events of the `quote_native` calculated in the fill event?
https://github.com/openbook-dex/openbook-v2/blob/36eb9c419994cb8064987a0176cb16d8a51b08fc/programs/openbook-v2/src/state/open_orders_account.rs#L127

If so, I think the rounding error is acceptable. Otherwise I'm afraid this change would be much more invasive since the taker fee can only be calculated after multiple fill events are emitted.

Reply:
exactly, the differences will be only attributable to the rounding in `market.taker_fees_ceil()`.  just wanted to make it clear because I wasn't sure how much precision you would need for the events; maybe i'd suggests to add the suffix `_ceil` to `taker_fess` or something similar to make it completely explicit in the code.

On the other hand, `total_quote_lots_taken` will be equal to the sum over all fill events `quote_native`, so you could approximate the self-trade amount with the difference between the actual fees paid & the hypothetical fee that would be paid if no self-trade amount was discounted.

Reply:
Thanks for the explanation!